### ISSUE TITLE
Problem about running keras on GPU.

### ISSUE THREAD

**Issue by Bjoux2** [fchollet/keras#7172]
[FAQ-How can I run Keras on GPU?]( https://keras.io/getting-started/faq/#how-can-i-run-keras-on-gpu) said: 

> If you are running on the TensorFlow or CNTK backends, your code will automatically run on GPU if any available GPU is detected.

I have two GPUs and the keras, TF, CUDA and cudnn have been successfully installed.

when I run[ mnist_cnn.py demo](https://github.com/fchollet/keras/blob/master/examples/mnist_cnn.py) on my computer, it gives: 

**Using TensorFlow backend.**
x_train shape: (60000, 28, 28, 1)
60000 train samples
10000 test samples
Train on 60000 samples, validate on 10000 samples
Epoch 1/12
2017-06-29 10:36:51.823568: W tensorflow/core/platform/cpu_feature_guard.cc:45] **The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.**
2017-06-29 10:36:51.823589: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-06-29 10:36:51.823593: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-06-29 10:36:51.823596: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-06-29 10:36:51.823599: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
    1/60000 [..............................] - ETA: 9419s - loss: 2.1261 - acc: 0.0000e+00
    3/60000 [..............................] - ETA: 4316s - loss: 2.3505 - acc: 0.0000e+00
    5/60000 [..............................] - ETA: 3303s - loss: 2.2857 - acc: 0.2000    
    7/60000 [..............................] - ETA: 2915s - loss: 2.2944 - acc: 0.1429
    9/60000 [..............................] - ETA: 2677s - loss: 2.3428 - acc: 0.1111
   11/60000 [..............................] - ETA: 2524s - loss: 2.3234 - acc: 0.0909
   13/60000 [..............................] - ETA: 2418s - loss: 2.3367 - acc: 0.0769
   15/60000 [..............................] - ETA: 2346s - loss: 2.3584 - acc: 0.0667
   17/60000 [..............................] - ETA: 2297s - loss: 2.3443 - acc: 0.0588
   19/60000 [..............................

Yes, it is running...

But I think it is running on CPUs rather than on GPUs, the reason is 
![default](https://user-images.githubusercontent.com/12781926/27669561-edd77c32-5cb9-11e7-862e-0f637bb3b6bb.png)

the usage of CPUs are really high while the useage of GPUs are very low (closed to 0).

How can I run keras on GPUs?

Thanks!

**Comment by mark86092** [comment]
You may check whether your tensorflow version is GPU version. I install it with `pip3 install tensorflow-gpu`

**Comment by Bjoux2** [comment]
@mark86092 
My tesorflow verison is GPU version.

thl@CHEM-INFO:~$ pip install tensorflow-gpu
Requirement already satisfied: tensorflow-gpu in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied: mock>=2.0.0 in /usr/local/lib/python2.7/dist-packages (from tensorflow-gpu)
Requirement already satisfied: numpy>=1.11.0 in /usr/local/lib/python2.7/dist-packages (from tensorflow-gpu)
Requirement already satisfied: protobuf==3.1.0 in /usr/local/lib/python2.7/dist-packages (from tensorflow-gpu)
Requirement already satisfied: wheel in /usr/lib/python2.7/dist-packages (from tensorflow-gpu)
Requirement already satisfied: six>=1.10.0 in /usr/lib/python2.7/dist-packages (from tensorflow-gpu)
Requirement already satisfied: funcsigs>=1; python_version < "3.3" in /usr/local/lib/python2.7/dist-packages (from mock>=2.0.0->tensorflow-gpu)
Requirement already satisfied: pbr>=0.11 in /usr/local/lib/python2.7/dist-packages (from mock>=2.0.0->tensorflow-gpu)
Requirement already satisfied: setuptools in /usr/local/lib/python2.7/dist-packages (from protobuf==3.1.0->tensorflow-gpu)

**Comment by mark86092** [comment]
In my experience of using tensorflow and keras, I think this problem may be a tensorflow problem, not a keras. 

In python, `import tensorflow`
![image](https://user-images.githubusercontent.com/1908007/27689517-03beead8-5d11-11e7-8aef-dc6ecefca57c.png)


```
>>> tensorflow.Session() 
```
something like this
![image](https://user-images.githubusercontent.com/1908007/27689661-7df28882-5d11-11e7-858f-595647f6949c.png)
